app/controllers/socure_webhook_controller.rb

```diff
@@ -10,17 +10,16 @@ class SocureWebhookController < ApplicationController
 
   def create
     log_webhook_receipt
-    fetch_results if socure_params.dig(:event, :eventType) == 'DOCUMENTS_UPLOADED'
-    render json: { message: 'Secret token is valid.' }
+    process_webhook_event
+    return render json: { message: 'Secret token is valid.' }, status: :ok
```

Comment:
seems like this could go inside an [ensure ](https://ruby-doc.org/docs/ruby-doc-bundle/UsersGuide/rg/ensure.html)